plotly/tools.py

```diff
@@ -1443,6 +1443,376 @@ class FigureFactory(object):
+            if ((type(index_vals[0]) == float) or (type(index_vals[0]) == int)):
+                for entry in index_vals:
+                    if not ((type(entry) == float) or (type(entry) == int)):
+                        raise exceptions.PlotlyError("Error in data"
+                                                     "frame. Make sure "
+                                                     "that all entries "
+                                                     "of each column are "
+                                                     "either numbers or "
+                                                     "strings.")
+                            
+            if (type(index_vals[0]) == str):
+                for entry in index_vals:
+                    if (type(entry) != str):
+                        raise exceptions.PlotlyError("Error in data"
+                                                     "frame. Make sure "
+                                                     "that all entries "
+                                                     "of each column are "
+                                                     "either numbers or "
+                                                     "strings.")
+
+        if useindex == False:
+            # Make list of column names
+            for name in df:
+                col_names.append(name)
+
+            # Populate matrix = [] with dataframe columns
+            for name in col_names:
+                matrix.append(df[name].values.tolist())
+
+
+            # Check if values in each column are either
+            # all strings or all numbers
+
+            # Matrix Check
+            for vector in matrix:   
+                if ((type(vector[0]) == float) or (type(vector[0]) == int)):
+                    for entry in vector:
+                        if not ((type(entry) == float) or (type(entry) == int)):
+                            raise exceptions.PlotlyError("Error in data"
+                                                         "frame. Make sure "
+                                                         "that all entries "
+                                                         "of each column are "
+                                                         "either numbers or "
+                                                         "strings.")
+                                
+                if (type(vector[0]) == str):
+                    for entry in vector:
+                        if (type(entry) != str):
+                            raise exceptions.PlotlyError("Error in dataframe. "
+                                                         "Make sure that all "
+                                                         "entries of each col"
+                                                         "umn are either "
+                                                         "numbers or strings.")
+
+        # Main Code
```

Comment:
Instead of using a `#Main Code` comment, you could consider splitting this into separate methods with clear method names. This is the way most of the other `FigureFactory` methods have been created.